pulp_2to3_migration/app/plugin/content.py

```diff
@@ -233,3 +364,108 @@ class RelatePulp2to3Content(Stage):
 
             for d_content in batch:
                 await self.put(d_content)
+
+
+class InterrelateContent(Stage):
+    """
+    Stage for relating Content to other Content.
+    """
+
+    async def run(self):
+        """
+        Relate each item in the input queue to objects specified on the DeclarativeContent.
+        """
+        async for dc in self.items():
+
+            if dc.extra_data.get('man_rel'):
+                self.relate_manifest_to_list(dc)
+            elif dc.extra_data.get('blob_rel'):
+                self.relate_blob(dc)
+            elif dc.extra_data.get('tag_rel'):
+                self.relate_manifest_tag(dc)
+
+            if dc.extra_data.get('config_blob_rel'):
+                self.relate_config_blob(dc)
+
+            await self.put(dc)
+
+    def relate_config_blob(self, dc):
+        """
+        Relate a Blob to a Manifest as a config layer.
+
+        Args:
+            dc (pulpcore.plugin.stages.DeclarativeContent): dc for a Blob
+        """
+        configured_dc_id = dc.extra_data.get('config_blob_rel')
+        # find blob by id
+        blob = Blob.objects.filter(digest=configured_dc_id).first()
+        dc.content.config_blob = blob
+        dc.content.save()
+
+
+    def relate_blob(self, dc):
+        """
+        Relate a Blob to a Manifest.
+
+        Args:
+            dc (pulpcore.plugin.stages.DeclarativeContent): dc for a Blob
+        """
+        related_dc_id_list = dc.extra_data.get('blob_rel')
+        blob_list = Blob.objects.filter(digest__in=related_dc_id_list)
+        for blob in blob_list:
+            thru = BlobManifest(manifest=dc.content, manifest_blob=blob)
+            thru.save()
+
+    def relate_manifest_tag(self, dc):
+        """
+        Relate an ImageManifest to a Tag.
+
+        Args:
+            dc (pulpcore.plugin.stages.DeclarativeContent): dc for a Tag
+        """
+
+        related_man_id = dc.extra_data.get('tag_rel')
+        man = Manifest.objects.filter(digest=related_man_id).first()
+        artifact = man._artifacts.get()
+        # add manifest's artifact
+        da = DeclarativeArtifact(
+                    artifact=artifact,
+                    url=NOT_USED,
+                    relative_path=dc.content.name,
+                    remote=NOT_USED,
+                    deferred_download=False)
+        dc.d_artifacts.append(da)
+        dc.content.tagged_manifest = man
+        # TODO fix tags
```

Comment:
busybox has: latest: schema2 and latest:schema1
alpine has: latest: shema2 and latest:schema1

in pulp3 we will have created only 2 Tags instead of 4 because of how model is define and we do not have tagged_manifest saved yet https://github.com/pulp/pulp_docker/blob/master/pulp_docker/app/models.py#L144

Reply:
solved by relating tag to manifest in the pre_save hook and setting does_batch=False on DC